connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorMaker.java

```diff
@@ -322,6 +324,29 @@ public void run() {
         }
     }
 
+    /**
+     * Validates that the underlying connectors and tasks are running.
+     * Visible for testing purpose.
+     */
+    boolean isConfiguredAndRunning() {
+        return CONNECTOR_CLASSES.stream().allMatch(connectorClazz -> {
+            final String connName = connectorClazz.getSimpleName();
+            return herderPairs.stream().allMatch(sourceAndTarget -> {
+                final ConnectorStateInfo connectorStatus = this.connectorStatus(sourceAndTarget, connName);
+                return connectorStatus != null
+                    // verify that connector state is set to running
+                    && connectorStatus.connector().state().equals(AbstractStatus.State.RUNNING.toString())
+                    // verify that all tasks are set to running
+                    && connectorStatus.tasks().stream().allMatch(s -> s.state().equals(AbstractStatus.State.RUNNING.toString()));
+            });
+        });
+    }
```

Comment:
We should expose the smallest testing-only APIs necessary in non-testing code, and implement other logic in testing-only places.

So in this case, the API we'd expose here would be the `connectorStatus` method, and the logic in this method would go in, e.g., the `DedicatedMirrorIntegrationTest` suite.